pkg/ccl/streamingccl/streamproducer/stream_lifetime.go

```diff
@@ -311,6 +320,24 @@ func buildReplicationStreamSpec(
 		return nil, err
 	}
 
+	parts := int(streamMaxProcsPerPartition.Get(&evalCtx.Settings.SV))
+	const minPartitionSpans = 10
+
+	repartitioned := make([]sql.SpanPartition, 0, len(spanPartitions)*parts)
+	for _, sp := range spanPartitions {
+		l := max((len(sp.Spans)/parts)+1, minPartitionSpans)
+		remaining := sp.Spans
+		for len(remaining) > 0 {
+			if len(remaining) > l {
+				sp.Spans, remaining = remaining[:l], remaining[l:]
+			} else {
+				sp.Spans, remaining = remaining, nil
+			}
+			repartitioned = append(repartitioned, sp)
+		}
+	}
+	spanPartitions = repartitioned
```

Comment:
if you fancied, you could pull this repartitioning logic into a helper func that could be unit testable. I worry that our unit and roachtests would not catch any regressions to this logic because they don't create enough partitions to test this logic.

Reply:
Done

Reply:
I also put a big comment on the helper summarizing our lunch convo.

Reply:
looks like this update didn't go through? also, you may need to update the commit message.